internal/audit/server_checks.go

```diff
@@ -230,3 +230,48 @@ func makeCheckServerResourceLimits(memoryUsageThreshold, storeUsageThreshold flo
 		return Pass, nil
 	}
 }
+
+func checkServerLeafnodeNames() checkFunc {
+	return func(r *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+		checkIfNameContainsSpaces := func(name string) bool {
+			for _, c := range name {
+				if c == ' ' {
+					return true
+				}
+			}
+			return false
+		}
+
+		for _, clusterName := range r.GetClusterNames() {
+			clusterTag := archive.TagCluster(clusterName)
+
+			for _, serverName := range r.GetClusterServerNames(clusterName) {
+				serverTag := archive.TagServer(serverName)
+
+				var serverLeafz server.Leafz
+				err := r.Load(&serverLeafz, clusterTag, serverTag, archive.TagServerLeafs())
+				if err != nil {
+					return Skipped, fmt.Errorf("failed to load Leafz for server %s: %w", serverName, err)
+				}
+
+				if len(serverLeafz.Leafs) == 0 {
+					continue
+				}
+
+				for _, leaf := range serverLeafz.Leafs {
+					// check if leaf name contains any spaces
+					if checkIfNameContainsSpaces(leaf.Name) {
+						examples.add("%s/%s: %s", clusterName, serverName, leaf.Name)
+					}
+				}
+
+			}
+			if examples.Count() > 0 {
+				logCritical("Found %d instances of servers with leafnode names containing spaces", examples.Count())
+				return Fail, nil
+			}
+		}
+		return Pass, nil
```

Comment:
To make this better I'd suggest creating a set of invalid leafnode names per cluster, adding that as a single example.

i.e. above:
```
if contains whitespace
  invalidLeafnodes[leafnodeName] = struct{}
```

at the end of the cluster loop:
```
if len(invalidLeafnodes) > 0 {
  examples.add("%s: %s", clusterName, invalidLeafnodes.keys())
}
```